### PR TITLE
feat(hermes): bridge Hermes Agent hook events into pipelock scanner pipeline

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary

- New `pipelock-hermes-hook` binary: a one-shot binary that reads a Hermes Agent hook event on stdin, runs pipelock's DLP + injection pipeline, and writes a `{"decision":"block","reason":"..."}` or `{}` JSON decision on stdout matching Hermes' documented shell-hook wire protocol.
- New `pipelock hermes install` subcommand: extracts the bundled Python plugin tree to `~/.hermes/plugins/pipelock/` with atomic-write + `.bak` rotation, so the same binary can be wired either via the in-process plugin (registers 5 hooks) or directly from `~/.hermes/config.yaml` under `hooks:`.
- Scanner reach: `pre_tool_call`, `transform_tool_result`, and `pre_gateway_dispatch` get DLP + injection scanning. `on_session_start` / `on_session_end` are observer-only.

## Why

Hermes Agent (`github.com/NousResearch/hermes-agent`, MIT) exposes a first-class plugin/hook system. MCP-only wrapping covers a single transport; the plugin/hook path also covers terminal, write_file, web extract, browser, and gateway inbound. This lets pipelock scan those surfaces at the tool boundary inside the agent process, independent of network reachability.

## Fail-closed behaviour

- Stdin larger than the 4 MiB payload cap → block
- Empty stdin / malformed JSON / missing `hook_event_name` → block
- Unknown `hook_event_name` → block (new event types must be explicitly classified; silent allow would bypass scanning)
- Config load failure → block
- Subprocess timeout, non-zero exit, or empty stdout (plugin's view) → block
- Secret encoded in a JSON object key → caught (key + value extraction)

## Out of scope (follow-ups)

- Backend-aware env injection for Hermes' terminal backends
- `pipelock hermes verify` / `pipelock hermes rollback`
- Cluster identity separation, file_sentry hard-block, Audit Packet receipt correlation

## Notes

Two pre-existing security-suite failures reproduce on `origin/main` and are unrelated to this change: the proxy adaptive-exempt red-team case and the `verify-install` dev-binary bundle-signature check.

## Summary by CodeRabbit

* **New Features**
  * Hermes Agent integration: installable plugin, Hermes CLI command, and a standalone hook CLI that evaluates hook events and returns allow/block decisions for tool, gateway, and session events; CLI accepts config and timeout.

* **Documentation**
  * Plugin README describing installation, hook protocol, fail-closed behavior, env overrides, and reinstall/backup semantics.

* **Tests**
  * Extensive tests covering hook CLI behavior, input limits and shapes, decision logic, install/extract flows, and filesystem edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/luckyPipewrench/pipelock/pull/605?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: cmd/pipelock-hermes-hook/main.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+			// Bundle merge errors are surfaced via stderr only; an empty
+			// bundle set still produces a valid (core-pattern-only)
+			// scanner. Blocking the whole hook on a bundle parse error
+			// would be a hostile-rules denial-of-service vector.
+			result := rules.MergeIntoConfig(c, cliutil.Version)
+			for _, e := range result.Errors {
+				_, _ = fmt.Fprintf(stderr, "pipelock-hermes-hook: warning: bundle %s: %s\n", e.Name, e.Reason)
+			}
+		},
+	})
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply cmd/pipelock-hermes-hook/main.go:150]: ResolveRuntime returns (*Config, ResolveRuntimeInfo), not (*Config, error). See internal/config/runtime.go:114. The discarded second value is resolution metadata, not an error, so there is no failure to drop. The suggested cfg, err = assignment would not compile. This matches how pipelock mcp scan consumes it at internal/cli/runtime/mcp.go:306.

File: internal/cli/hermes/plugin.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+			return nil
+		}
+		dest := filepath.Join(rootReal, relPath)
+		if err := ensureContained(rootReal, dest); err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			if err := os.MkdirAll(dest, pluginDirPerm); err != nil {
+				return fmt.Errorf("hermes: create %s: %w", dest, err)
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/cli/hermes/plugin.go:104]: The embedded plugin tree is flat: three files (__init__.py, plugin.py, README.md) directly under the root, with no subdirectories. The installer walk therefore never creates an intermediate directory that a symlinked component could hijack. Each write targets rootReal/<file> whose parent is the EvalSymlinks-resolved root, and ensureContained rejects any lexical escape. Per-component symlink resolution would harden a nested layout the template does not have. Worth revisiting if the plugin template ever gains subdirectories.

**Review by DeathCamel58 — APPROVED** [review]
